app/src/main/java/com/wasim/taskmaster/MainActivity.java

```diff
@@ -1,56 +1,87 @@
+import com.wasim.taskmaster.activities.AllTasks;
+import com.wasim.taskmaster.activities.SettingsActivity;
+import com.wasim.taskmaster.activities.TaskDetailActivity;
 
 public class MainActivity extends AppCompatActivity {
+    public static String Task_One;
     private final String TAG = "MainActivity";
 
+    SharedPreferences preferences;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
+        preferences = PreferenceManager.getDefaultSharedPreferences(this);
         setupAddTaskButton();
+        setupAllTaskButton();
+        setupSettingsButton();
+        setupTaskOneButton();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        setupRepopulateUsername();
+    }
 
+    void setupRepopulateUsername() {
+        preferences = PreferenceManager.getDefaultSharedPreferences((this));
+        String userName = preferences.getString(USERNAME_TAG, "Need Username");
+        TextView repopulateUsername = findViewById(R.id.MainActivityLabelTextView);
+        repopulateUsername.setText(userName + "'s Tasks");
+    }
 
+    void setupAllTaskButton() {
         Button allTasksButton = findViewById(R.id.AllTasks);
-        allTasksButton.setOnClickListener((new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-
-            }
-        }));
+        allTasksButton.setOnClickListener(view -> {
+            Intent goToAllTasksIntent = new Intent(MainActivity.this, AllTasks.class);
+            startActivity(goToAllTasksIntent);
+        });
     }
 
-void setupAddTaskButton(){
-    Button addTaskButton = findViewById(R.id.AddTaskButton);
+    void setupAddTaskButton() {
+        Button addTaskButton = findViewById(R.id.AddTaskButton);
+        addTaskButton.setOnClickListener(view -> {
+            Intent goToAddTasksIntent = new Intent(MainActivity.this, AddTask.class);
+            startActivity(goToAddTasksIntent);
+        });
+    }
 
-    addTaskButton.setOnClickListener(new View.OnClickListener() {
-        @Override
-        public void onClick(View view) {
-            System.out.println("Add Task Pressed");
-            Log.v(TAG,  "I'm a verbose log");
-            Log.d(TAG, "I'm a DEBUG log");
-            Log.i(TAG, "I;m an INFO log");
-            Log.w(TAG, "I'm a warning log");
-            Log.e(TAG, "I'm an error log");
-            Log.wtf(TAG, "What a terrible failure");
+    void setupSettingsButton() {
+        ImageView settingsButton = findViewById(R.id.mainActivitySettingsButton);
+        settingsButton.setOnClickListener(view -> {
+            Intent goToSettingsIntent = new Intent(MainActivity.this, SettingsActivity.class);
+            startActivity(goToSettingsIntent);
+        });
+    }
 
-            (( TextView)findViewById(R.id.MainActivityLabelTextView)).setText("");
+    void setupTaskOneButton() {
```

Comment:
Keep in mind that you'll be building an app with an unknown number of tasks. How could you modify this method to work with *any* task button?